pages/connect.tsx

```diff
@@ -1,61 +1,57 @@
-import { PermissionType } from 'arconnect';
-import React, { useState, useEffect } from 'react';
 import Router from 'next/router';
+import * as Styled from '../styles/connect';
+import React, { useState, useEffect } from 'react';
 import { useGetAllData } from '../hooks/useGetAllData';
-import { useGetUser } from '../hooks/useGetUser';
 import { UserProps } from '../types';
+import { useGetUser } from '../reducers/userContext';
 
 const Connect = () => {
-  const [userName, setUserName] = useState(''),
+  const [userExists, setUserExists] = useState(false),
+    [userName, setUserName] = useState(''),
     [userNames, setUserNames] = useState<string[]>([]),
-    { currentUser } = useGetUser(),
+    { user: currentUser, dispatch } = useGetUser(),
     { users } = useGetAllData();
 
   useEffect(() => {
     (async () => {
       try {
         const allUserNames = users.map((user: UserProps) => user.userName);
         setUserNames(allUserNames);
+        users.find((user) => {
```

Comment:
This code is highly ineffective.
This can be abstracted to 
```
let doesUserExist = users.some((user) => user.walletAddress === currentUser.walletAddress);
setUserExists(doesUserExist);
```